File: Java/Arrays/Insercion_final.java
/*
Arrays en Java,Insercion en el ultimo elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 04/09/2026
*/

import java.util.Arrays;

public class Insercion_final {
    public static void main(String[] args){
        int[] input_arr = {11, 21, 31, 41, 51, 61};     //array estatico

        System.out.println("Antes de la insercion, el array es: ");

        for (int elemento : input_arr){
            System.out.println(elemento);
        }
        System.out.println("\nDespues de la insercion, el array es: ");

        int[] new_input_arr = Arrays.copyOf(input_arr, input_arr.length + 1);       //otra forma de copiar un arreglo estatico
        new_input_arr[input_arr.length] = 67;       //agregamos el elemento final en el ultimo indice

        for (int elemento : new_input_arr){
            System.out.println(elemento);
        }
    }
}   

File: C++/Arrays/Insercion_final.cpp
/*
Arrays en C++,Insercion en el ultimo elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 06/09/2026
*/

#include <iostream>
#include <vector>

using namespace std;
int main()
{
    int input_arr[] = {1, 11, 21, 31, 41, 51};
    int tamano = sizeof(input_arr)/sizeof(input_arr[0]);
    int elem {61};

    cout << "Antes de la insercion, el array es: ";
    for (int elemento : input_arr){
        cout << elemento << " ";
    }

        //Nuevo arreglo
    int new_input_arr [tamano + 1];

    //Insercion
    for (int i = 0; i <= tamano; i++ ){
        new_input_arr[i] = input_arr[i];
    }
    new_input_arr[tamano] = elem;

    cout << "\nDespues de la insercion, el array es: ";
    for (int elemento : new_input_arr){
        cout << elemento << " ";
    }


    //Otra forma
    vector <int> other_arr = {1, 11, 21, 31, 41, 51};

    other_arr.insert(other_arr.begin() + tamano, elem);
    cout << "\n\nDespues de la insercion, el array es: ";
    for (int elemento : other_arr){
        cout << elemento << " ";
    }
    return 0;
}